platform/javascript/detect.py

```diff
@@ -46,38 +49,55 @@ def configure(env):
         # run-time performance.
         env.Append(CCFLAGS=['-Os'])
         env.Append(LINKFLAGS=['-Os'])
-        if env['target'] == 'release_debug':
-            env.Append(CPPDEFINES=['DEBUG_ENABLED'])
-            # Retain function names for backtraces at the cost of file size.
-            env.Append(LINKFLAGS=['--profiling-funcs'])
-    else:
+    elif env['target'] == 'release_debug':
+        env.Append(CCFLAGS=['-Os'])
+        env.Append(LINKFLAGS=['-Os'])
+        env.Append(CPPDEFINES=['DEBUG_ENABLED'])
+        # Retain function names for backtraces at the cost of file size.
+        env.Append(LINKFLAGS=['--profiling-funcs'])
+    else: # 'debug'
         env.Append(CPPDEFINES=['DEBUG_ENABLED'])
         env.Append(CCFLAGS=['-O1', '-g'])
         env.Append(LINKFLAGS=['-O1', '-g'])
         env.Append(LINKFLAGS=['-s', 'ASSERTIONS=1'])
 
-    ## Compiler configuration
+    if env['tools']:
+        if not env['threads_enabled']:
+            raise RuntimeError("Threads must be enabled to build the editor. Please add the 'threads_enabled=yes' option")
+        # Tools need more memory
+        env.Append(LINKFLAGS=['-s', 'TOTAL_MEMORY=33554432'])
```

Comment:
Is this 33 MB or 33 GB? Might be worth documenting where the number comes from.

Reply:
It's not documented in emscripten, like most of the stuff in there. It's mentioned, but never explained.
It's also going to be renamed in the upcoming version (so they can keep breaking stuff I guess).
I basically found out about it because the compilation failed for with a message about raising that value. There only explanation I found for it is in `settings.js`
https://github.com/emscripten-core/emscripten/blob/master/src/settings.js#L142

Reply:
Okay, I've added a small comment to it.

Reply:
I guess it's a number of bytes, so here 32 MiB (vs the default of 16 MiB).